File: algorithm/math/combinatorics/permutation/factorial/Factorial.java
import java.math.*;
import java.util.stream.*;

public class Factorial {

	public static int sample_factorial(int n) {
		return IntStream.rangeClosed(1, n)
			.reduce(1, (int a, int b) -> a * b);
	}

	public static BigInteger factorial(int n) {
		return IntStream.rangeClosed(1, n)
			.mapToObj(BigInteger::valueOf)
			.reduce(BigInteger.ONE, BigInteger::multiply);
	}
}

File: algorithm/math/combinatorics/permutation/factorial/factorial.c
#include <stdio.h>

int factorial(int n)
{
	int factorial = 1;
	while (n)
		factorial *= n--;

	return factorial;
}

int recur_factorial(int n)
{
	if (n == 0)
		return 1;

	return n * factorial(n - 1);
}